### DISCUSSION TITLE
What does the descriptor property of StartCircuit arguments represent?

### DISCUSSION THREAD

**Discussion by jababda** [dotnet/aspnetcore#25947] [General] [answered]
Context, we're getting an odd issue occurring in Chrome where after navigating to our server side blazor page, leaving the page for ~5 mins in the same tab and then returning.
Running Net Core 3.1 in IISExpress using windows auth with no anon access. 

The error being logged is:
`System.Security.Cryptography.CryptographicException: The payload expired at 16/09/2020 7:07:13 AM +00:00.
   at Microsoft.AspNetCore.DataProtection.TimeLimitedDataProtector.UnprotectCore(Byte[] protectedData, DateTimeOffset now, DateTimeOffset& expiration)
   at Microsoft.AspNetCore.DataProtection.TimeLimitedDataProtector.Unprotect(Byte[] protectedData, DateTimeOffset& expiration)
   at Microsoft.AspNetCore.DataProtection.TimeLimitedDataProtector.Microsoft.AspNetCore.DataProtection.IDataProtector.Unprotect(Byte[] protectedData)
   at Microsoft.AspNetCore.Components.Server.ServerComponentDeserializer.DeserializeServerComponent(ServerComponentMarker record)`

The only shared component between the requests is the value of **descriptor** in the logged message (Using Serilog)
`(Microsoft.AspNetCore.SignalR.Internal.DefaultHubDispatcher) Received hub invocation: InvocationMessage { InvocationId: "0", Target: "StartCircuit", Arguments: [ https://localhost:44352/dashboard/, https://localhost:44352/dashboard/, [{"type":"server","sequence":0,"descriptor":"<value here>"}] ], StreamIds: [  ] }.`

We get a different error in edge when we run the same scenario but much further into the application after we impersonate a windows user to make a web request to our api server so I suspect the problem stems from the browsers doing some odd caching of the windows auth that get provided via the descriptor value. 

I don't know if this is expected behavior or not but without understanding exactly what the descriptor property represents I don't think I can fix the route cause.

Any help or direction would be hugely appreciated!

**Comment by SteveSandersonMS** [comment, marked as answer]
This is interesting. Thanks for letting us know.

This is something we'd really want to investigate in case there's an underlying problem. Do you mind possible reposting this as an issue at https://github.com/dotnet/aspnetcore/issues? If you can give any more details about the repro steps it would help. For example what it means to "return" to the page after 3-5 mins (are you reloading the page? why does the browser take any action if you're just switching tabs?)

Also my first guess about the possible cause is caching in the browser. If the browser is caching the initial HTML response and then you try to restart the connection with an old expired connection descriptor, this sort of error is what I'd expect.

**Comment by SteveSandersonMS** [comment]
Hope you don't mind, but I'll close/lock this discussion so that we can continue when it's reposted as an issue. That's just to ensure other people don't get confused and come posting other things here :)
